R/tax_range_strat.R

```diff
@@ -0,0 +1,225 @@
+#' Generate a stratigraphic section plot
+#'
+#' A function to plot the stratigraphic ranges of fossil taxa from occurrence
+#' data.
+#'
+#' @param occdf \code{dataframe}. A dataframe of fossil occurrences containing
+#'   at least two columns: names of taxa, and their stratigraphic position (see
+#'   `name` and `level` arguments).
+#' @param name \code{character}. The name of the column you wish to be treated
+#'   as the input names, e.g. "genus" (default).
+#' @param level \code{character}. The name of the column you wish to be treated
+#'   as the stratigraphic levels associated with each occurrence, e.g. "bed"
+#'   (default) or "height". Stratigraphic levels must be \code{numeric}.
+#' @param certainty \code{character}. The name of the column you wish to be
+#'   treated as the information on whether an identification is certain (1) or
+#'   uncertain (0). By default (\code{certainty = NULL}), no column name is
+#'   provided, and all occurrences are assumed to be certain. In the plot,
+#'   certain occurrences will be plotted with a black circle and joined with
+#'   solid lines, while uncertain occurrences will be plotted with a white
+#'   circle and joined with dashed lines.
+#' @param by \code{character}. How should the output be sorted? Either: "FAD"
+#'   (first appearance; default), "LAD" (last appearance), or "name"
+#'   (alphabetically by taxon names).
+#' @param plot_args A list of optional arguments that are passed directly to
+#'   [graphics::plot()]. Useful arguments include `xlab` (the x-axis label),
+#'   `ylab` (the y-axis label, default is "Bed number"), `main` (the plot
+#'   title), `xlim` (the x-axis limits), and `ylim` (the y-axis limits). The
+#'   `axes` and `type` arguments are not supported and will be overridden.
+#' @param x_args A list of optional arguments that are passed directly to
+#'   [axis()] when generating the x-axis. Useful arguments include `font` (e.g.,
+#'   `3` is italic) and `las` (label orientation).
+#' @param y_args A list of optional arguments that are passed directly to
+#'   [axis()] when generating the y-axis. Useful arguments include `font` (e.g.,
+#'   `3` is italic) and `las` (label orientation).
+#'
+#' @return Invisibly returns a data.frame of the calculated taxonomic
+#'   stratigraphic ranges.
+#'
+#'   The function is usually used for its side effect, which is to create a plot
+#'   showing the stratigraphic ranges of taxa in a section, with levels at which
+#'   the taxon was sampled indicated with a point.
+#'
+#' @section Developer(s): Bethany Allen, William Gearty & Alexander Dunhill
+#' @section Reviewer(s): William Gearty & Lewis A. Jones
+#' @importFrom graphics axis par segments plot points box
+#'
+#' @examples
+#' #Load tetrapod dataset
```

Comment:
```suggestion
#' # Load tetrapod dataset
```
Just standardising...